pkg/vgmanager/vgmanager_controller.go

```diff
@@ -393,11 +441,83 @@ func (r *VGReconciler) addDevicesToVG(vgName string, devices []internal.BlockDev
 	return nil
 }
 
-// filterMatchingDevices returns unmatched and matched blockdevices
-// TODO: Implement this
-func filterMatchingDevices(blockDevices []internal.BlockDevice, volumeGroup *lvmv1alpha1.LVMVolumeGroup) ([]internal.BlockDevice, []internal.BlockDevice, error) {
-	// currently just match all devices
-	return []internal.BlockDevice{}, blockDevices, nil
+// filterMatchingDevices returns matched blockdevices
+func (r *VGReconciler) filterMatchingDevices(blockDevices []internal.BlockDevice, volumeGroup *lvmv1alpha1.LVMVolumeGroup) ([]internal.BlockDevice, error) {
+
+	var filteredBlockDevices []internal.BlockDevice
+
+	if volumeGroup.Spec.DeviceSelector != nil && len(volumeGroup.Spec.DeviceSelector.Paths) > 0 {
+		vgs, err := ListVolumeGroups(r.executor)
+		if err != nil {
+			return []internal.BlockDevice{}, fmt.Errorf("failed to list volume groups. %v", err)
+		}
+
+		for _, path := range volumeGroup.Spec.DeviceSelector.Paths {
+			diskName, err := filepath.EvalSymlinks(path)
+			if err != nil {
+				err = fmt.Errorf("unable to find symlink for disk path %s: %v", path, err)
+				return []internal.BlockDevice{}, err
+			}
+
+			isAlreadyExist := isDeviceAlreadyPartOfVG(vgs, diskName, volumeGroup)
+			if isAlreadyExist {
+				continue
+			}
+
+			blockDevice, ok := hasExactDisk(blockDevices, diskName)
+
+			if filepath.Dir(path) == internal.DiskByPathPrefix {
+				// handle disk by path here such as /dev/disk/by-path/pci-0000:87:00.0-nvme-1
+				if ok {
+					blockDevice.DiskByPath = path
+					filteredBlockDevices = append(filteredBlockDevices, blockDevice)
+				} else {
+					err = fmt.Errorf("can not find device path %s, device name %s in the available block devices", path, diskName)
+					return []internal.BlockDevice{}, err
+				}
+			} else if filepath.Dir(path) == internal.DiskByNamePrefix {
+				// handle disk by names here such as /dev/nvme0n1
+				if ok {
+					filteredBlockDevices = append(filteredBlockDevices, blockDevice)
+				} else {
+					err := fmt.Errorf("can not find device name %s in the available block devices", path)
+					return []internal.BlockDevice{}, err
+				}
+			} else {
+				err = fmt.Errorf("unsupported disk path format %s", path)
```

Comment:
please add to the error message that only `/dev/disk/by-path` and `/dev/` links are currently supported

Reply:
done